src/host/ft_host/CJK_DbcsTests.cpp

```diff
@@ -551,1143 +402,999 @@ void DbcsWriteRead::SendOutput(const HANDLE hOut,
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x029 | 0x0000 (0x00) | <null>
+    // 0x007 | 0x0000 (0x00) | <null>
+    // 0x007 | 0x0000 (0x00) | <null>
+    // 0x007 | 0x0000 (0x00) | <null>
+    // 0x007 | 0x0000 (0x00) | <null>
+    // ...
+    // "Space Padded" means most of the unused data in the buffer will be filled with spaces and the default attribute.
+    // "Dedupe" means that any full-width characters in the buffer (despite being stored doubled inside the buffer)
+    //    will be returned as single copies.
+    // "W" means that we intend Unicode data to be browsed in the resulting struct (even though wchar and char are unioned.)
+    // "Truncated" means that this pattern trims off some of the end of the buffer with NULLs.
+    // "A Cover Attr" means that after all the other operations, we will finally run through and cover up the attributes
+    //     again with what they would have been for multi-byte data (leading and trailing flags)
+    static constexpr CharInfoPattern TrueTypeCharANullWithAttrs{
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, colored),
+        makeCharInfo(0x0000, white),
+        makeCharInfo(0x0000, white),
+        makeCharInfo(0x0000, white),
+        makeCharInfo(0x0000, white),
+    };
 }
 
-void DbcsWriteRead::PrepReadConsoleOutput(_In_ const unsigned int uiCodePage,
-                                          _In_ PCSTR pszTestData,
-                                          const WORD wAttrOriginal,
-                                          const WORD wAttrWritten,
-                                          const DbcsWriteRead::WriteMode WriteMode,
-                                          const bool fWriteWithUnicode,
-                                          const bool fIsTrueTypeFont,
-                                          const bool fReadWithUnicode,
-                                          _Inout_updates_all_(cExpectedNeeded) CHAR_INFO* const rgciExpected,
-                                          const size_t cExpectedNeeded)
+const CharInfoPattern& DbcsWriteRead::PrepReadConsoleOutput(
+    const DbcsWriteRead::WriteMode WriteMode,
+    const UnicodeMode fWriteWithUnicode,
+    const bool fIsTrueTypeFont,
+    const bool fReadWithUnicode)
 {
     switch (WriteMode)

```

Comment:
I'm trusting that the logic in here still works basically the same. Pivoting the order on the conditionals here really made the diff hard to track, but if all the tests pass, then I'm pretty sure logic worked out the same